### DISCUSSION TITLE
Amending athinput values in pgen

### DISCUSSION THREAD

**Discussion by hwhitehead** [PrincetonUniversity/athena#579] [Q&A] [answered]
I could use some clarification on the startup chronology for the Mesh construction. Is it possible to change the dimensions of the Mesh as specified in the athinput file using the problem generator (specifically in InitUserMeshData)?

e.g. if the input file specifies `nx3 = 64` but also calls for the simulation to be 2D, could the problem generator override this to set `nx3 == 1`?

From a brief look over [main.cpp](https://github.com/PrincetonUniversity/athena/blob/master/src/main.cpp) I don't believe this is possible as the Mesh is constructed directly from the pinput object, but please correct me if it is wrong.

**Comment by tomidakn** [comment, marked as answer]
Sorry for my late response.

Technically you can override the input parameters through the ParameterInput class, but the grid structure (and some other parameters) cannot be modified. In particular, the Mesh is already created when Mesh::InitUserMeshData is called. The best you can do is output an error message and stop the simulation.

**Reply by hwhitehead** [reply]
Thanks, figured this was the case but good to have it verified.
